wasm/stats.cc

```diff
@@ -0,0 +1,125 @@
+#include <string>
+#include <unordered_map>
+
+#include "proxy_wasm_intrinsics.h"
+
+static bool isInbound()
+{
+  int64_t direction = 0;
+  getValue({"listener_direction"}, &direction);
+  return direction == 1;
+}
+
+using RqTotalCounter = Counter<std::string, std::string, std::string, std::string, std::string, std::string, std::string, std::string, std::string>;
+using RqDurationHist = Histogram<std::string, std::string, std::string, std::string, std::string, std::string, std::string, std::string>;
+
+class StatsContext : public Context
+{
+public:
+  explicit StatsContext(uint32_t id, RootContext *root) : Context(id, root),
+                                                          rq_total(RqTotalCounter::New("osm_request_total",
```

Comment:
I guess this will allow us to not tag the metrics at prometheus scrape time, as most of it will come from these.
Let me know when to test this, with so many labels with possible high cardinality values, i suspect this will choke prometheus to a rather fast yet painful death.

Reply:
I suppose we could add the `source_*` labels at scrape time since we're already only generating these metrics at the source. Not sure how we could work around the destination ones though, which are probably the bigger issue.